__init__.py

```diff
@@ -103,145 +34,52 @@ def handle_search(self, message):
-            data["related_queries"] = related_queries
-
-            if intent == "birthdate":
-                answer = infobox.get("born")
-
-            data["query"] = query
-            data["answer"] = answer
-            data["image"] = self.image
-        if not answer:
-            return None, None, None
-        answer = self.translate(answer)
-        return answer, level, data
-
     def CQS_match_query_phrase(self, utt):
         self.log.debug("DuckDuckGo query: " + utt)
-
-        answer, match, data = self.parse_subintents(utt)
-        if answer:
+        summary = self.ask_the_duck(utt)
+        if summary:
             self.idx += 1
-            return (utt, match, answer, data)
-
-        # extract most relevant keyword
-        utt = self.translate(utt, "en", self.lang)
-        keywords = self.rake.extract_keywords(utt)
-
-        self.log.debug("Extracted keywords: " + str(keywords))
-        # TODO better selection / merging of top keywords with same
-        #  confidence??
-        for kw in keywords:
-            query = kw[0]
-            self.log.debug("Selected keyword: " + query)
-
-            summary = self.ask_the_duck(query, translate=False)
-            if summary:
-                self.idx += 1
-                return (utt, CQSMatchLevel.GENERAL, self.results[0],
-                        {'query': query, 'answer': self.results[0],
-                         "keywords": keywords, "image": self.image})
+            self.get_image(utt)
+            return (utt, CQSMatchLevel.GENERAL, self.results[0],
+                    {'query': utt,
+                     'answer': self.results[0],
+                     "image": self.image})
 
     def CQS_action(self, phrase, data):
         """ If selected show gui """
-        print(data)
-        self.display_ddg(data["answer"], data["image"])
-
-    # duck duck go api
-    def ask_the_duck(self, query, translate=True):
-        if translate:
-            # Automatic translation to English
-            utt = self.translate(query, "en", self.lang)
-        else:
-            utt = query
-
-        # cache so we dont hit the api twice for the same query
-        if query not in self.duck_cache:
-            self.duck_cache[query] = requests.get("https://api.duckduckgo.com",
-                                                  params={"format": "json",
-                                                          "q": utt}).json()
-        data = self.duck_cache[query]
-
-        # GUI
-        self.gui.clear()  # clear previous answer just in case
-        title = data.get("Heading")
-        self.image = data.get("Image", "")
-
-        # summary
-        summary = data.get("AbstractText")
-
-        if not summary:
-            return None
+        self.display_ddg(data["answer"])
 
-        self.log.debug("DuckDuckGo answer: " + summary)
+    def get_image(self, query):
+        data = self.duck.get_data(query, {})
+        self.image = data.get("Image") or self.image
```

Comment:
Should `self.image` associate by query or user? This implementation won't work for serving multiple clients

Reply:
this skill does not support multiple users, the "tell me more" intents have global state. image should match the global state